tests/test_detect_target.py

```diff
@@ -5,21 +5,55 @@
 
 import cv2
 import numpy as np
+import pathlib
 import pytest
 import torch
 
 from modules.detect_target import detect_target
 from modules import image_and_time
+from modules import detections_and_time
 
 
 DEVICE =                        0 if torch.cuda.is_available() else "cpu"
-MODEL_PATH =                    "tests/model_example/yolov8s_ultralytics_pretrained_default.pt"
-OVERRIDE_FULL =                 False  # Tests are able to handle both full and half precision.
-IMAGE_BUS_PATH =                "tests/model_example/bus.jpg"
-IMAGE_BUS_ANNOTATED_PATH =      "tests/model_example/bus_annotated.png"
-IMAGE_ZIDANE_PATH =             "tests/model_example/zidane.jpg"
-IMAGE_ZIDANE_ANNOTATED_PATH =   "tests/model_example/zidane_annotated.png"
+MODEL_PATH =                    pathlib.Path("tests", "model_example", "yolov8s_ultralytics_pretrained_default.pt")
+OVERRIDE_FULL =                 not torch.cuda.is_available()  # CPU does not support half precision
+IMAGE_BUS_PATH =                pathlib.Path("tests", "model_example", "bus.jpg")
+BOUNDING_BOX_BUS_PATH =         pathlib.Path("tests", "model_example", "bounding_box_bus.txt")
+IMAGE_ZIDANE_PATH =             pathlib.Path("tests", "model_example", "zidane.jpg")
+BOUNDING_BOX_ZIDANE_PATH =      pathlib.Path("tests", "model_example", "bounding_box_zidane.txt")
 
+BOUNDING_BOX_DESIRED_PRECISION = 7
+CONFIDENCE_DESIRED_PRECISION = 3
+
+def compare_detections(actual: detections_and_time.DetectionsAndTime, expected: detections_and_time.DetectionsAndTime) -> None:
+    """
+    Compare expected and actual detections.
+    """
+    assert len(expected.detections) == len(actual.detections)
+
+    for i in range(0, len(expected.detections)):
+        expected_detection = expected.detections[i]
+        actual_detection = actual.detections[i]
+
+        assert expected_detection.label == actual_detection.label
+        np.testing.assert_almost_equal(expected_detection.confidence, actual_detection.confidence, decimal=CONFIDENCE_DESIRED_PRECISION)
+
+        np.testing.assert_almost_equal(expected_detection.x1, actual_detection.x1, decimal=BOUNDING_BOX_DESIRED_PRECISION)
+        np.testing.assert_almost_equal(expected_detection.y1, actual_detection.y1, decimal=BOUNDING_BOX_DESIRED_PRECISION)
+        np.testing.assert_almost_equal(expected_detection.x2, actual_detection.x2, decimal=BOUNDING_BOX_DESIRED_PRECISION)
+        np.testing.assert_almost_equal(expected_detection.y2, actual_detection.y2, decimal=BOUNDING_BOX_DESIRED_PRECISION)
+
+def create_detections(expected: np.ndarray) -> detections_and_time.DetectionsAndTime:
```

Comment:
Rename the variable to something like `detections_from_file` , since it is in an encoded format.